src/serverenvironment.h

```diff
@@ -257,6 +257,19 @@ class ServerEnvironment : public Environment
 	*/
 	u16 addActiveObject(ServerActiveObject *object);
 
+	/**
+	 * Verify if id is a free active object id
+	 * @param id
+	 * @return true if slot is free
+	 */
+	bool isFreeServerActiveObjectId(u16 id) const;
+
+	/**
+	 * Retrieve the next free activeobject ID
+	 * @return free activeobject ID or zero if not free ID found
```

Comment:
`Retrieve the first free ActiveObject ID`
`@return free ActiveObject ID or 0 if none was found`